app/Listeners/LogInToWiki.php

```diff
@@ -37,6 +37,11 @@ public function __construct(Request $request, UserCreator $mediawikiUserCreator)
      */
     public function handle(Login $event): void
     {
+        // Early return if wiki integration is disabled or dependencies are missing
+        if (env('FEATURE__WIKI_INTEGRATION') !== true || $this->wikiUserCreator === null) {
```

Comment:
I think we can simplify a lot of these checks, but I'm fine being explicit. i.e.:

```php
        if (!env('FEATURE__WIKI_INTEGRATION') || !$this->wikiUserCreator) {

```